build.gradle

```diff
@@ -37,8 +39,8 @@ ext {
 	linkScmDevConnection = 'git@github.com:spring-projects/spring-amqp.git'
 	springAsciidoctorBackendsVersion = '0.0.7'
 
-	modifiedFiles =
-			files(grgit.status().unstaged.modified).filter{ f -> f.name.endsWith('.java') || f.name.endsWith('.kt') }
+	modifiedFiles = providers.provider {
+			files(grgit.status().unstaged.modified).filter{ f -> f.name.endsWith('.java') || f.name.endsWith('.kt') }}
```

Comment:
In my experience in other projects this has to be like this:
```
    modifiedFiles =
            files()
                    .from {
                        files(grgit.status().unstaged.modified)
                                .filter { f -> f.name.endsWith('.java') || f.name.endsWith('.kt') }
                    }
    modifiedFiles.finalizeValueOnRead()
```

Lazy, once calculated on demand and cached.